lib/grpc/server.ex

```diff
@@ -161,12 +161,16 @@ defmodule GRPC.Server do
          req
        ) do
     last = fn r, s ->
-      reply = apply(server, func_name, [r, s])
-
-      if res_stream do
-        {:ok, stream}
-      else
-        {:ok, stream, reply}
+      case apply(server, func_name, [r, s]) do
```

Comment:
This is interesting, but it is a breaking change. I think some tests are needed so we can move forward with this PR